lms/djangoapps/course_api/courseware_access.py

```diff
@@ -0,0 +1,109 @@
+""" Centralized access to LMS courseware app """
+
+from opaque_keys import InvalidKeyError
+from opaque_keys.edx.keys import CourseKey, UsageKey
+from xmodule.modulestore.django import modulestore
+from xmodule.modulestore.exceptions import ItemNotFoundError
+
+from courseware import courses, module_render
+from courseware.model_data import FieldDataCache
+
+
+def get_course(course_id, depth=0):
+    """
+    Utility method to obtain course components
+    """
+    course_descriptor = None
+    course_content = None
+    course_key = CourseKey.from_string(course_id)
+    if course_key:
+        course_descriptor = get_course_descriptor(course_key, depth)
+    return course_descriptor, course_key, course_content
+
+
+def get_course_child(request, user, course_key, content_id, depth=None):
+    """
+    Return a course xmodule/xblock to the caller
+    """
+    child_descriptor = None
+    child_content = None
+    child_key = get_course_child_key(content_id)
+    if child_key:
+        child_descriptor = get_course_child_descriptor(child_key, depth=depth)
+        if child_descriptor:
+            child_content = get_course_child_content(request, user, course_key, child_descriptor)
+    return child_descriptor, child_key, child_content
```

Comment:
Only child_descriptor is used by the single caller of this function. This is a small enough amount of code that I think inlining it would be reasonable.